www-client/uget: version bump to 2.2.0

someone-stole-my-name opened the pull request:
- Bump to latest version (2.2.0 18/1/13).
- Bump to EAPI 6
- Drop autotools and eutils
- Drop flags that no longer exist
- Added functions to handle icons
- Added myself as a proxy-maintainer (not sure if that is all is needed)

I had to make 2 commits...apparently cp is too hard for me, I copied the file from the wrong overlay. Any criticism and advice is welcome since this is one of my firsts ebuilds.

mgorny commented on www-client/uget/metadata.xml:
```diff
@@ -12,6 +20,10 @@
 			You'll find the plugin in the app's settings.</flag>
 		<flag name="hide-temp-files">Make temporary files that are used while downloading hidden.</flag>
 		<flag name="gtk3">Use x11-libs/gtk+:3 instead of x11-libs/gtk+:2</flag>
+		<flag name="appindicator">Add support for AppIndicator</flag>
```
Please sort lexically, and preferably keep one style (`"` vs `'`).

mgorny commented on www-client/uget/uget-2.2.0.ebuild:
```diff
@@ -0,0 +1,62 @@
+KEYWORDS="~amd64 ~arm ~x86"
+SRC_URI="mirror://sourceforge/urlget/${P}.tar.gz"
+
+DESCRIPTION="Download manager using gtk+ and libcurl"
```
Please order variables as in skeleton files as it is otherwise hard to read different ebuilds, i.e.

```
DESCRIPTION
HOMEPAGE
SRC_URI

LICENSE
SLOT
KEYWORDS
IUSE
REQUIRED_USE

RDEPEND
DEPEND

EVERYTHING_ELSE
```

mgorny commented on www-client/uget/uget-2.2.0.ebuild:
```diff
@@ -0,0 +1,62 @@
+	"
+
+src_configure() {
+	econf $(use_enable nls) \
```
For such a long list, I would suggest doing it like this (non-obligatory):

```
local myconf=(
    $(use_with ...)
    ...
)

econf "${myconf[@]}"
```

Then you don't need to play with backslashes.

mgorny commented on www-client/uget/uget-2.2.0.ebuild:
```diff
@@ -0,0 +1,62 @@
+}
+
+pkg_preinst() {
+	gnome2_icon_savelist
```
This is no longer necessary.

mgorny commented on www-client/uget/uget-2.2.0.ebuild:
```diff
@@ -0,0 +1,62 @@
+	gstreamer? ( media-libs/gstreamer:1.0 )
+	libnotify? ( x11-libs/libnotify )
+	aria2? ( net-misc/aria2[xmlrpc] )
+	appindicator? ( dev-libs/libappindicator:= )
```
I'm pretty sure you need to depend on specific slot here, not `:=`.

mgorny commented on www-client/uget/uget-2.2.0.ebuild:
```diff
@@ -0,0 +1,62 @@
+	dev-libs/libpcre
+	>=dev-libs/glib-2.32:2
+	>=x11-libs/gtk+-3.4:3
+	gnutls? ( net-libs/gnutls dev-libs/libgcrypt:= )
```
Slot `:11` doesn't provide headers, so you want `:0=`. Also please don't list multiple deps in a single line.

mgorny commented on www-client/uget/uget-2.2.0.ebuild:
```diff
@@ -0,0 +1,62 @@
+DOCS=( AUTHORS ChangeLog NEWS README )
+
+REQUIRED_USE="^^ ( gnutls openssl )"
+RDEPEND="
```
Please sort everything lexically.

mgorny commented on www-client/uget/uget-2.2.0.ebuild:
```diff
@@ -0,0 +1,62 @@
+LICENSE="LGPL-2.1"
+SLOT="0"
+
+DOCS=( AUTHORS ChangeLog NEWS README )
```
Is this different than the result of installing via default list?

someone-stole-my-name replied:
NEWS is not installed by default.

mgorny replied:
Are you sure about that? It's there in the code.

mgorny commented on www-client/uget/metadata.xml:
```diff
@@ -12,6 +20,10 @@
 		<flag name="gtk3">Use x11-libs/gtk+:3 instead of x11-libs/gtk+:2</flag>
+		<flag name="appindicator">Add support for AppIndicator</flag>
+		<flag name='openssl'>Use <pkg>dev-libs/openssl</pkg> instead of <pkg>net-libs/gnutls</pkg></flag>
+		<flag name='sockets'>Enable JSON-RPC over unix domain socket</flag>
```
To be honest, this flag name looks wrong. 'Sockets' make me think of all kinds of sockets, not just AF_UNIX.

someone-stole-my-name replied:
I committed most of the changes already but this one. Which one do you recommend? I went with sockets because of gpsd but control-socket is used by clsync to enable a similar feature, should that be used instead?

mgorny replied:
What is the purpose of this socket, exactly? (you may also want to include this in description)

someone-stole-my-name replied:
It enables a JSON-RPC interface, I'm not sure what the use case is (maybe some 3rd party gui?) or how many users actually use this feature.

mgorny commented on www-client/uget/metadata.xml:
```diff
@@ -1,17 +1,29 @@
 		category.
 	</longdescription>
 	<use>
+		<flag name="appindicator">Add support for AppIndicator</flag>
```
That's global USE=ayatana, actually:

```
 global:ayatana: Build in support for Ayatana notification using the libindicate or libappindicator plugin
```

someone-stole-my-name commented:
Last commit addresses that issue and also renames sockets to control-socket since sockets is also a global use that has nothing to do with this feature.

mgorny commented:
Could you confirm that it doesn't work correctly without explicit `DOCS`? Also either squash that into one commit, or I'll do that for you.

someone-stole-my-name commented:
No it doesn't but maybe is because some of the files are 0byte dummies, I just removed DOCS altogether and rebased... The important docs do get installed. Thank you for your patience 😃

mgorny commented:
Thanks. Merged it now, and reassigned the bugs to you. Please close them as appropriate.

```
  KEYWORDS.dropped              1
   www-client/uget/uget-2.2.0.ebuild: ppc
```

Please file a re-keywording request for that.

```
  dependency.missingslot        2
   www-client/uget/uget-1.10.4.ebuild: RDEPEND: 'dev-libs/libgcrypt' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
   www-client/uget/uget-9999.ebuild: RDEPEND: 'dev-libs/libgcrypt' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
```

Please look into fixing it.

```
  inherit.deprecated            6
[...]
   www-client/uget/uget-9999.ebuild: please migrate from 'git-2' to 'git-r3' on line: 11
```

Please either update the live ebuild or remove it.
